src/libstore/build/local-derivation-goal.cc

```diff
@@ -1454,7 +1463,8 @@ void LocalDerivationGoal::stopDaemon()
         thread.join();
     daemonWorkerThreads.clear();
 
-    daemonSocket = -1;
+    // release the socket.
+    daemonSocket.release();
```

Comment:
This would leak the file descriptor if it hasn't been closed yet.

Reply:
If `close` is idempotent, we could just close fore good measure here prior to the release. I'll look at the code.